include/xrpl/protocol/Permissions.h

```diff
@@ -0,0 +1,82 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2024 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#ifndef RIPPLE_PROTOCOL_PERMISSION_H_INCLUDED
+#define RIPPLE_PROTOCOL_PERMISSION_H_INCLUDED
+
+#include <optional>
+#include <string>
+#include <unordered_map>
+
+namespace ripple {
+
+/**
+ * We have transaction type permissions and granular type
+ * permissions. Since we will reuse the TransactionFormats to parse the
+ * Transaction Permissions, we only define the GranularPermissionType here.
+ */
+
+enum GranularPermissionType : std::uint32_t {
+    gpTrustlineAuthorize = 65537,
+
+    gpTrustlineFreeze = 65538,
+
+    gpTrustlineUnfreeze = 65539,
+
+    gpAccountDomainSet = 65540,
+
+    gpAccountEmailHashSet = 65541,
+
+    gpAccountMessageKeySet = 65542,
+
+    gpAccountTransferRateSet = 65543,
+
+    gpAccountTickSizeSet = 65544,
+
+    gpPaymentMint = 65545,
+
+    gpPaymentBurn = 65546,
+
+    gpMPTokenIssuanceLock = 65547,
+
+    gpMPTokenIssuanceUnlock = 65548,
+};
+
+class Permission
```

Comment:
should probably delete copy and copy assignment constructors

Reply:
does this still needs to be done?

Reply:
just added